sbi/samplers/importance/importance_sampling.py

```diff
@@ -0,0 +1,128 @@
+from math import sqrt
+from typing import Tuple
+
+import torch
+from torch import Tensor
+
+
+def importance_sample(
+    potential_fn,
+    proposal,
+    num_samples: int = 1,
+) -> Tuple[Tensor, Tensor]:
+    """Returns samples from proposal and log(importance weights).
+
+    Args:
+        potential_fn: Unnormalized potential function.
+        proposal: Proposal distribution with `.sample()` and `.log_prob()` methods.
+        num_samples: Number of samples to draw.
+
+    Returns:
+        Samples and logarithm of importance weights.
+    """
+    samples = proposal.sample((num_samples,))
+
+    potential_logprobs = potential_fn(samples)
+    proposal_logprobs = proposal.log_prob(samples)
+    log_importance_weights = potential_logprobs - proposal_logprobs
+
+    return samples, log_importance_weights
+
+
+def exponentiate_weights(log_weights: Tensor) -> Tensor:
+    """Subtracts the maximum of the `log_weights` and then exponentiates them.
```

Comment:
it also filters for `inf`, so the input and output shape can differ, no? 
I think it would be good to add this to the docstring.